lib/libconfig.c

```diff
@@ -608,6 +608,13 @@ static void __config_write_setting(const config_t *config,
   char nongroup_assign_char = config_get_option(
     config, CONFIG_OPTION_COLON_ASSIGNMENT_FOR_NON_GROUPS) ? ':' : '=';
 
+  if(setting->comment)
+  {
+    if(depth > 1)
+      __config_indent(stream, depth, config->tab_width);
+    fprintf(stream, "// %s\n", setting->comment);
```

Comment:
If the comment contains newlines, this will result in a syntax error in the generated file. The comment should be written out such that each line begins with //.